WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -212,6 +213,58 @@ private extension ProductDetailPreviewViewModel {
     }
 }
 
+// MARK: - Site settings
+//
+private extension ProductDetailPreviewViewModel {
+    func fetchSettingsIfNeeded() async {
+        guard weightUnit == nil || dimensionUnit == nil else {
```

Comment:
❓ Since weight unit and dimension unit belong to product settings, I suppose we are fetching general settings for currency? Should we split this method into 2 parts: fetch general settings if currency is missing, and product settings if the units are missing?

Reply:
Good question. The currency is not an optional value like `weightUnit` or `dimensionUnit`. It looks like we are assigning `USD` as its default value.

So, I decided to refresh the currency if we find that a shipping unit is `nil`.